### DISCUSSION TITLE
Announcement/Notification Save/Restore State

### DISCUSSION THREAD

**Discussion by 80081** [music-assistant/support#1279] [Feature requests and ideas]
Back when I was using LMS, someone created a little automation/script thing called LMS TTS Notifications, sadly for me it stopped working pretty quickly so I deleted it.  The basic jist was that it made a service available for announcements, either TTS or media or both, but it saved the state before the call, so playlist, speaker group synchronisation, volume, track position etc.  Then after the announcement it restored the state.  I think a feature of this nature would be really useful, so we can use MA for doorbells and other notifications etc.  I realise that this could probably be achieved already, or at least will be possible once the missing slim CLI commands are implemented, but it is a lot cleaner as a service rather than manually implementing it all in service calls.

**Comment by OzGav** [comment]
This is already a feature of MA but it is not a service call it is just how it works. I haven’t tried it lately with the latest beta but it should work.

**Reply by 80081** [reply]
I'll check this when I get home, I recall trying it the other day and it replaced the playlist, but I'll double check.

**Reply by OzGav** [reply]
There are some issues with the queue in the current beta that are being addressed

**Reply by 80081** [reply]
So tried this today using google_say, ideally I'd like to be able to join players together for an announcement and then for the previous groups to be restored, but I can't test that as the slim cli commands aren't ready.  Anyway, test with google_say stopped the music played the announcement, it seems the playlist stayed intact, but the music didn't resume on it's own I had to press play manually and it started from the beginning of the track.  So in a nutshell announcements work out of the box better than they did on LMS but still could be improved as a feature.  Volume as a for instance if I wanted to make an announcement at a certain level it would have to be done seperately and restored seperately, rather than just a service that can handle all the details of an announcement.

**Reply by OzGav** [reply]
Yes the HA integration still has some functionality to be added. Please be patient as what you are after is coming and is a core feature.

**Reply by 80081** [reply]
> Yes the HA integration still has some functionality to be added. Please be patient as what you are after is coming and is a core feature.

I appologise if I seemed impatient, it was simply a suggestion for a feature, i'm fully aware of the work that's going into this and I'm grateful.  I'm not 100% sure what you're saying here though, I know that once the missing cli commands are implemented this would be possible by chaining commands together.  My suggestion was to have a bespoke service call that could handle all of these things as parameters.  So are you saying that once certain things are implemented it will just do this stuff out of the box?

**Reply by marcelveldt** [reply]
correct, I think/hope it will be available by the end of the week.

**Comment by zox-gh** [comment]
Any update on Announcement feature? Is it in the pipeline, at least.
I've just tested it with the latest MA release but no luck. It says "Call-service error. Music Assistant does not yet support announcements"

**Reply by OzGav** [reply]
Just keep an eye on the backlog. You can see what is being worked on and what is coming up. https://github.com/orgs/music-assistant/projects/2
